kedro/framework/cli/jupyter.py

```diff
@@ -31,20 +32,54 @@
 OVERWRITE_HELP = """If Python file already exists for the equivalent notebook,
 overwrite its contents."""
 
+jupyter_command_group_order = ["init", "notebook", "lab", "convert"]
+jupyter_command_group_order = {command: order for order, command in enumerate(jupyter_command_group_order)}
+
+
+class JupyterCommandGroup(click.Group):
+    """A custom class for ordering the `kedro jupyter` command groups"""
+
+    def __init__(self, name=None, commands=None, **attrs):
+        super().__init__(name, commands, **attrs)
+        #: the registered subcommands by their exported names.
+        self.commands = commands or {}
+
+    def list_commands(self, ctx):
+        """List commands according to a custom order"""
+        return sorted(self.commands, key=lambda k: jupyter_command_group_order[k])
+
```

Comment:
This allows us to inject a custom order. Initially I think about adding something like a `@deprecated` flag and which will automatically lower the rank of the commands.

However, I found that this is not worth the effort. Looking at the `kedro` command itself we may need to customize the order (It's alphabetically sorted now). 

Instead of having Customised Order + deprecated command move to the end, a simple list of order can be enough.

Reply:
I really like this!!